### PR TITLE
[SYCL] [FPGA] Leverage `add_ir_annotations_member` in latency controls

### PR DESCRIPTION
**Pull request by shuoniu-intel**
Clean up the experimental latency control headers by leveraging `add_ir_annotations_member`.
No change to the user APIs.

FPGA BE support: https://github.com/intel-innersource/applications.fpga.oneapi.products.acl-source/pull/1609

Local E2E tests combining this change and the FPGA BE change:
- On LSUs: https://psg-sc-arc.sc.intel.com/p/psg/swip/w/shuoniu/git/shuoniu.latency-control-prototype/add_ir_annotations_member/build/lsu.prj/reports/report.html
- On blocking pipes: https://psg-sc-arc.sc.intel.com/p/psg/swip/w/shuoniu/git/shuoniu.latency-control-prototype/add_ir_annotations_member/build/blocking_pipe.prj/reports/report.html
- On non-blocking pipes: https://psg-sc-arc.sc.intel.com/p/psg/swip/w/shuoniu/git/shuoniu.latency-control-prototype/add_ir_annotations_member/build/non_blocking_pipe.prj/reports/report.html

### CODE REVIEW

**Comment by shuoniu-intel** [comment]
Add @steffenlarsen to review the properties part.

File: sycl/include/sycl/ext/intel/experimental/fpga_lsu.hpp [hunk 1]
```diff
@@ -101,32 +85,16 @@ template <class... _mem_access_params> class lsu final {
...
-    if constexpr (_control_type == latency_control_type::exact) {
-      _control_type_code = 1;
-    } else if constexpr (_control_type == latency_control_type::max) {
-      _control_type_code = 2;
-    } else if constexpr (_control_type == latency_control_type::min) {
-      _control_type_code = 3;
+    _T *P = Ptr;
+    if constexpr (!std::is_same_v<
+                      _propertiesT,
+                      oneapi::experimental::detail::empty_properties_t>) {
```
Comment (steffenlarsen) [inline sycl/include/sycl/ext/intel/experimental/fpga_lsu.hpp:91]: While we are at it, we might as well make use the new extension alias.

```suggestion
                      oneapi::experimental::empty_properties_t>) {
```

**Review by steffenlarsen — COMMENTED** [review]
Just a few small ones.

File: sycl/include/sycl/ext/intel/experimental/fpga_lsu.hpp (hunk 1 above)
Comment (shuoniu-intel) [reply sycl/include/sycl/ext/intel/experimental/fpga_lsu.hpp]: Addressed

File: sycl/include/sycl/ext/intel/experimental/fpga_utils.hpp [hunk 2]
```diff
@@ -31,28 +32,22 @@ struct _GetValue<_Type, _T1, _T...> {
...
-  using type = DefaultPropValT;
-};
-template <typename PropListT, typename PropKeyT, typename DefaultPropValT>
-struct GetOrDefaultValT<
-    PropListT, PropKeyT, DefaultPropValT,
-    std::enable_if_t<PropListT::template has_property<PropKeyT>()>> {
-  using type = decltype(PropListT::template get_property<PropKeyT>());
+template <typename T, typename PropertyListT =
+                          ext::oneapi::experimental::detail::empty_properties_t>
+class annotated {
```
[the remark this answers is not in the document]
Comment (shuoniu-intel) [reply sycl/include/sycl/ext/intel/experimental/fpga_utils.hpp]: Addressed

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Review by KseniyaTikhomirova — APPROVED** [review]
LGTM

**Comment by shuoniu-intel** [comment]
@intel/llvm-gatekeepers could you help merge?

**Pull request merged**